jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/PluginConfigurationProcessor.java

```diff
@@ -188,14 +192,32 @@ static PluginConfigurationProcessor processCommonConfiguration(
     }
 
     if (projectProperties.isWarProject()) {
-      return JavaEntrypointConstructor.makeDistrolessJettyEntrypoint();
+      return null;
     }
 
     String mainClass = projectProperties.getMainClass(jibExtension);
     return JavaEntrypointConstructor.makeDefaultEntrypoint(
         AbsoluteUnixPath.get(parameters.getAppRoot()), parameters.getJvmFlags(), mainClass);
   }
 
+  /**
+   * Compute the container program arguments. If the entrypoint is not inherited, program arguments
+   * must not be inherited .
+   *
+   * @param entrypoint the container entrypoint
+   * @param jibExtension the {@link JibExtension} providing the configuration data
+   * @return the program arguments
+   */
+  @Nullable
+  static List<String> computeProgramArguments(
+      @Nullable List<String> entrypoint, JibExtension jibExtension) {
+    if (entrypoint != null && jibExtension.getContainer().getArgs() == null) {
```

Comment:
I believe this is duplicating the logic in `BuildImageSteps` - I think we could just use `jibExtension.getContainer().getArgs()` directly.

Reply:
> But I have to change `JavaDockerContextGenerator#makeDockerfile` to add `CMD []` if `ENTRYPOINT` is not null.

The correct way should be to not print anything about `CMD`, because if there exists `ENTRYPOINT` which will override the base value, Docker build will automatically reset CMD to null. (And null CMD is different from an empty list CMD.) But the problem we have here, I think (correct me if I'm getting something wrong, haven't looked into the overall code in detail), is that for the image built from our generated `Dockerfile` to be consistent with the image we build directly, we also have to reset CMD in the container metadata instead of setting it to an empty list.  And that seems a bit tricky for now: https://github.com/GoogleContainerTools/jib/pull/1071#discussion_r223760143 Not sure what would be the right/best way.

Reply:
But I didn't change this behaviour. 
`CMD []` was already there before this PR if we set only the entrypoint.

Reply:
> But I haven't change this behaviour. CMD [] was already there before this PR if we set only the entrypoint.

Sure, that was and will be the behavior that we explicitly override our `ENTRYPOINT` and `CMD` for non-WAR projects. In this case, we will always override `CMD` to `[]`. I'm talking about the case (currently this can apply to WAR projects) where the user is overriding `ENTRYPOINT` and left `CMD` undefined, in which case, `CMD` should be set to null instead of `[]`.

Reply:
Actually, since we are enabling Jib to inherit ENTRYPOINT and CMD, I think we should eventually change the behavior for the non-WAR case too, when we find a way to fix #924 which will allow users to inherit either ENTRYPOINT or CMD for non-WAR. After fixing #924, I imagine that the same rule for inheriting ENTRYPOINT and CMD will apply to both non-WAR and WAR.

Reply:
I looked into the code, and updated my previous comments in this particular conversation about the CMD being null vs. an empty list.